packages/libwasmer/libwasmer.3.0.0~beta+macos-x86_64/opam

```diff
@@ -0,0 +1,23 @@
+opam-version: "2.0"
+
+authors: ["wasmer"]
+maintainer: "engineering@wasmer.io"
+license: "MIT"
+homepage: "https://wasmer.io/"
+
+doc: "https://docs.wasmer.io/"
+bug-reports: "https://github.com/wasmerio/wasmer-ocaml/issues"
+dev-repo: "git+https://github.com/wasmerio/wasmer-ocaml.git"
+tags: ["org:wasmer"]
+
+synopsis: "The official Wasmer library"
+description: """
+This package is used by the wasmer package to trigger the installation
+of the Wasmer library."""
+
+extra-source "libwasmer.tar.gz" {
+	src: "https://github.com/wasmerio/wasmer/releases/download/3.0.0-beta/wasmer-darwin-amd64.tar.gz"
```

Comment:
The use of binary packages in opam is discouraged. You can see for example `tezos-rust-libs.1.0.0` as an example of a rust library built in opam by simply vendoring its dependencies and overriding the config file.

Reply:
This should work now.

Reply:
Update: it seems the system's Cargo and Rust are not up to date (Wasmer requires Rust 1.59 for now, while the system installs cargo 0.47.0 with rustc version 1.48.0), which makes the compilation fail ("feature `resolver` is required")... So I either need to execute the `rustup` script from rust-lang.org (`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`) -- which also requires an online connection during build, use a precompiled Rust binary (though I don't even know where to find the required version, let alone use it), or use the precompiled binaries.
I'm going back to using the precompiled binaries for now, but I'm open for alternatives.

(PS: this is how wasmtime does it as well, see #17776 and #17948.)

Reply:
> the system's Cargo and Rust are not up to date

which system? If it is only some of the distribution we're testing in CI but some worked it's totally fine.

Reply:
I think it was the 4.02 compiler in the CI? I'm not sure, but it used Debian 11.
In any case, some in the company want to push toward even more recent `rustc` ("the Rust ecosystem makes it so that using an outdated Rust version is near useless"), so using the package manager's `rustc` would be very unstable depending on the version.

Reply:
That's ok, some packages do work this way